core/shared/src/main/scala/com/dwolla/tracing/syntax/EntryPointRootScope.scala

```diff
@@ -1,54 +1,106 @@
-                           L: Local[F, Span[F]]): F[A] =
+                           L1: Local[F, Span[F]],
+                           L2: Logger[F],
+                          ): F[A] =
     runInContinuation(name, kernel, Span.Options.Defaults)(fa)
 
   def runInContinuation[A](name: String, kernel: Kernel, options: Span.Options)
                           (fa: F[A])
                           (implicit F: MonadCancelThrow[F],
-                           L: Local[F, Span[F]]): F[A] =
-    entryPoint
-      .continue(name, kernel, options)
-      .use(Local[F, Span[F]].scope(fa))
+                           L1: Local[F, Span[F]],
+                           L2: Logger[F],
+                          ): F[A] =
+    safeSpan(fa)(_.continue(name, kernel, options))
+
+  @deprecated("use variant accepting Logger[F]", "0.2.5")
+  private[syntax] def runInContinuationOrElseRoot[A](name: String, kernel: Kernel, fa: F[A], F: MonadCancelThrow[F], L: Local[F, Span[F]]): F[A] =
+    runInContinuationOrElseRoot(name, kernel, Span.Options.Defaults)(fa)(F, L, NoOpLogger(F))
+
+  @deprecated("use variant accepting Logger[F]", "0.2.5")
+  private[syntax] def runInContinuationOrElseRoot[A](name: String, kernel: Kernel, options: Span.Options, fa: F[A], F: MonadCancelThrow[F], L: Local[F, Span[F]]): F[A] =
+    safeSpan(fa)(_.continueOrElseRoot(name, kernel, options))(F, L, NoOpLogger(F))
 
   def runInContinuationOrElseRoot[A](name: String, kernel: Kernel)
-                           (fa: F[A])
-                           (implicit F: MonadCancelThrow[F],
-                            L: Local[F, Span[F]]): F[A] =
+                                    (fa: F[A])
+                                    (implicit F: MonadCancelThrow[F],
+                                     L1: Local[F, Span[F]],
+                                     L2: Logger[F],
+                                    ): F[A] =
     runInContinuationOrElseRoot(name, kernel, Span.Options.Defaults)(fa)
 
   def runInContinuationOrElseRoot[A](name: String, kernel: Kernel, options: Span.Options)
-                           (fa: F[A])
-                           (implicit F: MonadCancelThrow[F],
-                            L: Local[F, Span[F]]): F[A] =
-    entryPoint
-      .continueOrElseRoot(name, kernel, options)
+                                    (fa: F[A])
+                                    (implicit F: MonadCancelThrow[F],
+                                     L1: Local[F, Span[F]],
+                                     L2: Logger[F],
+                                    ): F[A] =
+    safeSpan(fa)(_.continueOrElseRoot(name, kernel, options))
+
+  private def safeSpan[A](fa: F[A])
+                         (f: EntryPoint[F] => Resource[F, Span[F]])
+                         (implicit F: MonadCancelThrow[F],
+                          L1: Local[F, Span[F]],
+                          L2: Logger[F]): F[A] =
+    Resource.applyFull { (poll: Poll[F]) =>
```

Comment:
This is the main bit; the rest of this is tests and preserving binary compatibility with previous releases.